h2o-algos/src/main/java/hex/DataInfo.java

```diff
@@ -1091,7 +1091,8 @@ public final int getCategoricalId(int cid, int val) {
     int [] offs = fullCatOffsets();
     int expandedVal = val + offs[cid];
     if(expandedVal >= offs[cid+1]) {  // previously unseen level
-      assert _valid : "Categorical value out of bounds, got " + val + ", next cat starts at " + fullCatOffsets()[cid + 1];
+      if (!isIWV || _useAllFactorLevels) // assertion only needed for non-interaction columns or _useAllFactorLevels=true
```

Comment:
why not merge with the assert? i.e.: assert (isIWV && !_useAllFactorLevels) || _isValid

Reply:
or move into the if branch bewow